packages/rancher-logging/generated-changes/overlay/templates/loggings/kube-audit/logging.yaml

```diff
@@ -0,0 +1,100 @@
+{{- if .Values.additionalLoggingSources.kubeAudit.enabled }}
+{{- $auditLogDir := default "/var/log/kube-audit" .Values.additionalLoggingSources.kubeAudit.logDirectory }}
+apiVersion: logging.banzaicloud.io/v1beta1
+kind: Logging
+metadata:
+  name: {{ .Release.Name }}-kube-audit
+  namespace: {{ .Release.Namespace }}
+  labels:
+{{ include "logging-operator.labels" . | indent 4 }}
+spec:
+  controlNamespace: {{ .Release.Namespace }}
+  {{- if .Values.additionalLoggingSources.kubeAudit.loggingRef }}
+  loggingRef: {{ .Values.additionalLoggingSources.kubeAudit.loggingRef }}
+  {{- end }}
+  disableKubernetesFilter: true
+  fluentbit:
+    image:
+      repository: {{ template "system_default_registry" . }}{{ .Values.images.fluentbit.repository }}
+      tag: {{ .Values.images.fluentbit.tag }}
+    inputTail:
+      Tag: "kube-audit"
+      Path: {{ $auditLogDir }}/audit-log.json
+      Parser: json
```

Comment:
This should probably be adjustable in `values.yaml`

Reply:
Could we default this based on provider and allow for user override?

Reply:
Sure, k3s has no default audit logging turned on at all, you have to manually pass the kube-api flags through, so I don't think a default makes sense for that.  I will check what the rke2 behaviour is. Rke is set in the cluster.yaml; is there an accepted way to pull values from there?

Reply:
I've refactored the helper I added and added some more to support this.  Also to clean things up I've just hardcoded the destination for the extra mount to `/kube-audit-logs` and then we can pick the file up from there.

For rke2 I've added the default location for a CIS hardened cluster - audit logging is automatically configured as part of the hardened config.

For k3s I've made the values required as there's no way to set a default for that.

Reply:
Discussed with @dbason offline, but we can assume that we will always know k3s/rke/rke2 with `additionalLoggingSources`. We always know with aks/gke/eks as well, but those are _optional_ via the dashboard checkbox. Thus, it can be a bit confusing (@paynejacob: maybe we should split cloud into its own section in the future?).

We should just use `additionalLoggingSources.$RANCHER_DISTRO.enabled` where possible.